### ISSUE TITLE
gstreamer: Plugins are not found

### ISSUE THREAD

**Issue by fuzzy-id** [NixOS/nixpkgs#10559]
After installing `gstreamer` and `gst_plugins_good` running `gst-inspect-0.10` doesn't show the additional plugins.

Anyway, one can set `GST_PLUGIN_PATH` to e.g. `$HOME/.nix-profile/lib/gstreamer-0.10` and the plugins are properly loaded.

**Comment by jagajaga** [comment]
Wasn't this fixed yet?

**Comment by fpletz** [comment]
Doesn't work for me on master.

**Comment by vcunat** [comment]
I think it's questionable whether this is desirable. We've had a similar discussion about python packages. In the case of similar plugins there's also the possibility that the version combination might make them incompatible; upstream packages typically assume that the packaging system ensures consistency, so this may result in ugly failures.

**Comment by mdorman** [comment]
It seems like we still need _some_ solution to this; we're shipping stuff that *can't* work:

```
mdorman@aching:~$ gst-play-1.0 foo
Failed to create 'playbin' element. Check your GStreamer installation.
```
That's with all the plugin packages and everything installed.  Add to the environment, though:

```
mdorman@aching:~$ GST_PLUGIN_SYSTEM_PATH_1_0=~/.nix-profile/lib/gstreamer-1.0 gst-play-1.0 foo
Press 'k' to see a list of keyboard shortcuts.
Now playing /home/mdorman/foo
ERROR Resource not found. for file:///home/mdorman/foo
ERROR debug information: gstfilesrc.c(530): gst_file_src_start (): /GstPlayBin:playbin/GstURIDecodeBin:uridecodebin0/GstFileSrc:source:
No such file "/home/mdorman/foo"
Reached end of play list.
```

**Comment by cx405** [comment]
I have same thing here, 17.09.

gst-play-1.0 foo
Failed to create 'playbin' element. Check your GStreamer installation.

I am also affected by this bug: #27050, which was closed but the problem still happening.

I noticed that most people with this bug have 2+ soundcards. Does it apply here? I am so frustrated that I am thinking about going gentoo..

**Comment by Dan-Theriault** [comment]
Similar thing here, 18.03.

I want to use mopidy with pulseaudio. The usual way of doing this is to put something like:
`[Audio]
sink=pulsesink`
int the mopidy configuration.

This gives me the following errors (from the service journal):
`Apr 07 00:19:02 specere mopidy[20142]: WARNING  Found fatal audio configuration errors:
Apr 07 00:19:02 specere mopidy[20142]: WARNING    audio/sink unknown config key.`

pulsesink is included in the pulseaudio gstreamer plugin, [which appears in the gst-plugins-good list here](https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-good-plugins/html/).
gst-plugins-good is a buildInput of mopidy.

If Nixpkgs is going to treat gstreamer like python, there should be the `gstreamer.withPlugins` analogue to `python.withPackages`. The current behavior is just plain broken.

**Comment by cx405** [comment]
I want to report that my issue was not this bug, but bug in KDE plugin system #27050, which was only present in 17.09 release (not present in 18.03) and for 17.09 the community helped me craft a workaround.

It seems that the bug applies to gst-play-1.0, and it needs a wrapper to actually see the environment, as in the post of @mdorman.

**Comment by bachp** [comment]
I was experimenting with `gst-inspect-1.0` on current master (197c393c2fd098ff46c722481c3b5576c38a4ba9). And it only dsiplays 1 plugin:

```
gst-inspect-1.0
staticelements:  bin: Generic bin
staticelements:  pipeline: Pipeline object

Total count: 1 plugin, 2 features
``` 

I installed all of the following in my `environment.systemPackages`:

```
gst_all_1.gstreamer
gst_all_1.gstreamer.dev 
gst_all_1.gst-plugins-base 
gst_all_1.gst-plugins-good 
gst_all_1.gst-plugins-bad 
gst_all_1.gst-plugins-ugly 
gst_all_1.gst-libav
```

What I would expect is for `gst-inspect-1.0` from `gst_all_1.gstreamer.dev ` to list all the plugins.

Also setting `GST_PLUGIN_SYSTEM_PATH_1_0=~/.nix-profile/lib/gstreamer-1.0` doesn't help as `~/.nix-profile/lib/gstreamer-1.0` doesn't even exist.

Is there something I'm missing?

**Comment by bachp** [comment]
I think this was fixed in #54340
